hs_core/tasks.py

```diff
@@ -355,8 +355,10 @@ def create_temp_zip(resource_id, input_path, output_path, aggregation_name=None,
         if '/data/contents/' in input_path:
             short_path = input_path.split('/data/contents/')[1]  # strip /data/contents/
             res.create_aggregation_xml_documents(path=short_path)
+            res.create_model_program_meta_json_schema_files(path=short_path)
         else:  # all metadata included, e.g., /data/*
             res.create_aggregation_xml_documents()
+            res.create_model_program_meta_json_schema_files()
```

Comment:
as mentioned before, I see no reason why we can't just create the json when xml is created.  They're the same thing, just a different metadata formats.